src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -555,28 +543,24 @@ protected function resetRequest(Application $app): void
             $method = self::CLEAN_CLEAR;
         }
 
-        switch ($method) {
-            case self::CLEAN_FORCE_RECREATE:
-            case self::CLEAN_RECREATE:
-                $app->set('request', $app->getComponents()['request']);
-                break;
-            case self::CLEAN_CLEAR:
+        match ($method) {
+            self::CLEAN_FORCE_RECREATE, self::CLEAN_RECREATE => $app->set('request', $app->getComponents()['request']),
+            self::CLEAN_CLEAR => (function () use ($request): void {
                 $request->getHeaders()->removeAll();
                 $request->setBaseUrl(null);
                 $request->setHostInfo(null);
-                $request->setPathInfo(null);
-                $request->setScriptFile(null);
-                $request->setScriptUrl(null);
-                $request->setUrl(null);
+                $request->setPathInfo('');
+                $request->setScriptFile('');
+                $request->setScriptUrl('');
+                $request->setUrl('');
```

Comment:
These calls are not the same actually. The code hints in Yii are wrong, for example function body:
```php
    /**
     * Sets the path info of the current request.
     * This method is mainly provided for testing purpose.
     * @param string $value the path info of the current request
     */
    public function setPathInfo($value)
    {
        $this->_pathInfo = $value === null ? null : ltrim($value, '/');
    }

```

One solution is to use a stub file. Here's the one I use in my Yii2 projects: https://gist.github.com/SamMousa/65bca72252cecde5144e3a82138333d1